src/parser_utils/package_json.ml

```diff
@@ -7,18 +7,27 @@
 
 module Ast = Flow_ast
 
+type flow_t = {
+  typesVersions: string SMap.t;
+}
+
 type t = {
   name: string option;
   main: string option;
+  flow: flow_t option;
 }
 
 type 'a t_or_error = (t, 'a * string) result
 
 let (>>=) = Core_result.(>>=)
 
-let empty = { name = None; main = None; }
+let empty = { name = None; main = None; flow = None; }
+
 let name package = package.name
 let main package = package.main
+let flow_types_versions package = match package.flow with
+  | Some t -> t.typesVersions
+  | None -> SMap.empty
```

Comment:
Is there any case where a client may want to distinguish between a package with no `flow` entry, a package with a `flow` entry that does not contain a `typesVersion` property, and a package with a `flow` entry containing an empty `typesVersion` property?

Reply:
I think there isn't any difference, but packages with empty `types_versions` shouldn't be added to cache and skip checking entirely, I think I missed this

Reply:
It also depends if we add more fields to `flow`, like #2675